xpdacq/calib.py

```diff
@@ -78,6 +79,117 @@ def run_calibration(exposure=60, calibrant_file=None, wavelength=None,
     gaussian : int, optional
         gaussian width between rings, Default is 100.
     """
+
+    _check_obj(_REQUIRED_OBJ_LIST)
+    ips = get_ipython()
+    bto = ips.ns_table['user_global']['bt']
+    prun = ips.ns_table['user_global']['prun']
+
+    # d-spacing
+    if calibrant_file is not None:
+        calibrant_name = os.path.split(calibrant_file)[1]
+        calibrant_name = os.path.splitext(calibrant_name)[0]
+    else:
+        calibrant_name = 'Ni'
+
+    # scan
+    calib_collection_uid = str(uuid.uuid4())
+    calibration_dict = {'sample_name':calibrant_name,
+                        'sample_composition':{calibrant_name :1},
+                        'is_calibration': True,
+                        'calibration_collection_uid': calib_collection_uid}
+    sa = Sample(bto, calibration_dict)
+    prun_uid = prun(calibration_dict, ScanPlan(bto, ct, exposure))
+    light_header = glbl.db[-1]
+    if dark_sub:
+        dark_uid = light_header.start['sc_dk_field_uid']
+        dark_header = glbl.db[dark_uid]
+        dark_img = np.asarray(glbl.db.get_images(dark_header,
+                                glbl.det_image_field)).squeeze()
+    for ev in glbl.db.get_events(light_header, fill=True):
+        img = ev['data'][glbl.det_image_field]
+        if dark_sub:
+            img -= dark_img
+
+    print('{:=^20}'.format("INFO: you are able to calib, please refer"
+                           "to guid here:\n"))
+    print('{:^20}'
+          .format("http://xpdacq.github.io/usb_Running.html#calib-manual"))
+    print()
+    # calibration, return a azimuthal integrator
+    ai = calibration(img, calibrant_file=calibrant_file,
+                     calib_collection_uid=calib_collection_uid,
+                     wavelength=wavelength, detector=detector,
+                     gaussian=gaussian)
+
+    # masking
+    print('INFO: create mask')
+    mask = get_mask(img, ai, glbl.mask_dict, save_name=None)
+
+    return ai
+
+
+def get_mask(img, geometry_object, mask_dict, save_name=None):
+    """ function to generate mask
+
+    Parameters
+    ----------
+    img : ndarray
+        image that going to be masked
+    geometry_object : pyFAI.geometry.Geometry
+        The pyFAI description of the detector orientation or any
+        subclass of pyFAI.geometry.Geometry class
+    save_name : str, optional
+        full path for this mask going to be saved. if it is None, 
+        only mask will be returned, no file will be saved locally.
+    mask_dict : dict, optional
+        dictionary for arguments in masking function. for more details, 
+        please see docstring of xpdan.tools.mask_img
+
+    See also
+    --------
+    xpdan.tools.mask_img
+    """
+    mask = mask_img(img, geometry_object, **mask_dict)
+    print("INFO: add mask to global state")
+    glbl.mask = mask
+
+    if save_name is not None:
```

Comment:
yes, no file will be saved, only glbl attribute is updated. I put this behavior as we are going to save the mask when we run ``integrate_and_save`` on ``xpdAn`` side.